cpp/src/arrow/compute/api_vector.h

```diff
@@ -246,6 +249,70 @@ class ARROW_EXPORT CumulativeSumOptions : public FunctionOptions {
   bool check_overflow = false;
 };
 
+/// \brief Options for cumulative product function
+class ARROW_EXPORT CumulativeProductOptions : public FunctionOptions {
+ public:
+  explicit CumulativeProductOptions(double start = 1, bool skip_nulls = false,
+                                    bool check_overflow = false);
+  explicit CumulativeProductOptions(std::shared_ptr<Scalar> start,
+                                    bool skip_nulls = false, bool check_overflow = false);
+  static constexpr char const kTypeName[] = "CumulativeProductOptions";
+  static CumulativeProductOptions Defaults() { return CumulativeProductOptions(); }
+
+  const bool is_minmax = false;
+  const bool is_max = false;
+
+  /// Optional starting value for cumulative product
+  std::shared_ptr<Scalar> start;
+
+  /// If true, nulls in the input are ignored and produce a corresponding null output.
+  /// When false, the first null encountered is propagated through the remaining output.
+  bool skip_nulls = false;
+
+  /// When true, returns an Invalid Status when overflow is detected
+  bool check_overflow = false;
+};
+
+/// \brief Options for cumulative min functions
+class ARROW_EXPORT CumulativeMinOptions : public FunctionOptions {
+ public:
+  explicit CumulativeMinOptions(bool skip_nulls = false);
+  explicit CumulativeMinOptions(double start, bool skip_nulls = false);
+  explicit CumulativeMinOptions(std::shared_ptr<Scalar> start, bool skip_nulls = false);
+  static constexpr char const kTypeName[] = "CumulativeMinOptions";
+  static CumulativeMinOptions Defaults() { return CumulativeMinOptions(); }
+
+  const bool is_minmax = true;
```

Comment:
nit: it might be more idiomatic to have this as `constexpr static bool`, then below use `if (OptionsType::is_minmax)`